libslide/src/evaluator_rules/rule.rs

```diff
@@ -57,6 +97,23 @@ impl PatternMap {
         }
         bootstrapped
     }
+
+    /// Checks a `PatternMap` is resolvable, returning an error if it is not.
+    pub fn validate(&self) -> Option<UnresolvedMapping> {
+        let unresolved_pats: Vec<Rc<_>> = unique_pats(&self.to)
+            .difference(&unique_pats(&self.from))
+            .map(|p| Rc::clone(*p))
+            .collect();
+
+        if unresolved_pats.is_empty() {
+            return None;
+        }
+
+        Some(UnresolvedMapping {
+            map: self.clone(),
+            unresolved_pats,
+        })
+    }
 }
```

Comment:
where does this return an error?

Reply:
See the return type and exactly these lines. UnresolvedMap is the error.